bgpd/bgp_vty.c

```diff
@@ -3600,6 +3600,53 @@ DEFUN (bgp_default_shutdown,
 	return CMD_SUCCESS;
 }
 
+DEFPY (bgp_shutdown_msg,
+       bgp_shutdown_msg_cmd,
+       "bgp shutdown message MSG...",
+       BGP_STR
+       "Enable administrative shutdown of the BGP instance\n"
+       "Add a shutdown message (RFC 8203)\n"
+       "Shutdown message\n")
+{
+	char* msgstr = NULL;
+
+	VTY_DECLVAR_CONTEXT(bgp, bgp);
+
+	if (argc > 3)
+		msgstr = argv_concat(argv, argc, 3);
+
+	bgp_shutdown_enable(bgp, msgstr);
+	XFREE(MTYPE_TMP, msgstr);
+
+	return CMD_SUCCESS;
+}
+
+DEFPY (bgp_shutdown, bgp_shutdown_cmd,
+       "bgp shutdown",
+       BGP_STR
+       "Enable administrative shutdown of the BGP instance\n")
+{
+	VTY_DECLVAR_CONTEXT(bgp, bgp);
+
+	bgp_shutdown_enable(bgp, NULL);
+
+	return CMD_SUCCESS;
+}
+
+DEFPY (no_bgp_shutdown,
+       no_bgp_shutdown_cmd,
+       "no bgp shutdown",
```

Comment:
Is there a specific reason to do that? Because peers won't be notified about lifting the shutdown, only about the shutdown itself. Or do you think it would be nice to be able to set a message to write to the log?

Reply:
added in latest commit [1b6e7a8](https://github.com/FRRouting/frr/pull/6938/commits/1b6e7a8874bcd7c5a48d2c849e64b0dd67f49f7e).